app/src/main/java/com/spoony/spoony/presentation/main/MainScreen.kt

```diff
@@ -0,0 +1,76 @@
+package com.spoony.spoony.presentation.main
+
+import android.app.Activity
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableLongStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.navigation.compose.NavHost
+import com.spoony.spoony.presentation.explore.navigation.exploreNavGraph
+import com.spoony.spoony.presentation.main.component.MainBottomBar
+import com.spoony.spoony.presentation.map.navigaion.mapNavGraph
+import com.spoony.spoony.presentation.register.navigation.registerNavGraph
+import kotlinx.collections.immutable.toPersistentList
+
+@Composable
+fun MainScreen(
+    navigator: MainNavigator = rememberMainNavigator()
+) {
+    val context = LocalContext.current
+
+    var backPressedTime by remember {
+        mutableLongStateOf(0L)
+    }
+
+    BackHandler(enabled = true) {
+        if (System.currentTimeMillis() - backPressedTime <= 2000L) {
+            (context as Activity).finish()
+        } else {
+            // TODO: 스낵바 자리
+        }
+        backPressedTime = System.currentTimeMillis()
+    }
+
+    Column(
+        modifier = Modifier.fillMaxSize(),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center
+    ) {
+        Scaffold(
+            bottomBar = {
+                MainBottomBar(
+                    visible = navigator.shouldShowBottomBar(),
+                    tabs = MainTab.entries.toPersistentList(),
+                    currentTab = navigator.currentTab,
+                    onTabSelected = {
+                        navigator.navigate(it)
+                    }
```

Comment:
```suggestion
                    onTabSelected = navigator::navigate
```
P5: 별건 아니지만.. 이런것도 있어요!

Reply:
별건 아니지 않은 거 같아요! 이 경우에서 메서드 레퍼런스 형태가 람다 생성 오버헤드를 줄여서 이방법이 좋은 것 같아요!